src/pool/srv_target.c

```diff
@@ -80,16 +80,29 @@ ds_pool_child_lookup(const uuid_t uuid)
 
 	d_list_for_each_entry(child, &tls->dt_pool_list, spc_list) {
 		if (uuid_compare(uuid, child->spc_uuid) == 0) {
-			child->spc_ref++;
 			return child;
 		}
 	}
 	return NULL;
 }
 
 struct ds_pool_child *
-ds_pool_child_get(struct ds_pool_child *child)
+ds_pool_child_get(const uuid_t uuid)
```

Comment:
How about adding a static `pool_child_lookup_noref` (or `ds_pool_child_lookup_noref` if you need an external one) and keeping the semantics of `ds_pool_child_lookup` unchanged (except for checking the state)? And we could remove `ds_pool_child_get` if necessary.

That seems to have less impact on other modules, and also more consistent with other "lookup/get/put" APIs we have.

Reply:
Ok, sounds good to me. I'll fix it.